### PR TITLE
fix(bundle): cache traces

### PR DESCRIPTION
**Pull request by JeroenVinke**
This PR is the same as https://github.com/aurelia/cli/pull/419, but squashed and it includes a fix for the mentioned A -> B -> CSS bug. 

cc @huochunpeng

### CODE REVIEW

**Comment by JeroenVinke** [comment]
@AStoker I don't think we did a thorough code review of this yet, could you take a look?

**Comment by AStoker** [comment]
I'll take another look around. @ahmedshuhel, think you could take a look at it too (or if you know somebody else who would be more familiar)? 
Also, props where props are due. @huochunpeng put a ton of work into this as well :) Thanks guys!

**Comment by JeroenVinke** [comment]
Definitely, @huochunpeng put a lot of time and effort into this one 👏

File: lib/build/amodro-trace/lib/loader/Loader.js [hunk 1]
```diff
@@ -352,66 +355,77 @@ function __exec(contents, r, d, c) {
             layer.existingRequireUrl = url;
           }
         } else {
-          // Load the file contents, process for conditionals, then
-          // evaluate it.
-          contents = context.fileRead(moduleName, url);
+          traceCacheEntry = getOwn(context._traceCache, moduleName);
+          if (traceCacheEntry && falseProp(context.needFullExec, moduleName)) {
```
Comment (AStoker) [inline lib/build/amodro-trace/lib/loader/Loader.js]: So just as a note, this `falseProp` check should be the only difference between master amodrotrace and our take on it.

Comment (huochunpeng) [inline lib/build/amodro-trace/lib/loader/Loader.js]: Looks great to me!

**Comment by AStoker** [comment]
@EisenbergEffect, if you're good with this, it would be great to have in the next release.

**Comment by EisenbergEffect** [comment]
Will do :)

**Comment by fabioluz** [comment]
Nice guys!

**Comment by EisenbergEffect** [comment]
Thank you everyone! It's merged.

**Comment by huochunpeng** [comment]
Finally:)

**Comment by fabioluz** [comment]
Yeah! I've just checked it and the building time blazingly faster! Well done guys! ✨ 👏 👏

**Comment by swalters** [comment]
Thanks to all involved.  This was the CLI commit of the year!!

**Pull request merged**